### PR TITLE
fix instance removing while profile changes

### PR DESCRIPTION
**Pull request by promiteyka**
Fix for https://github.com/yandex-qatools/teamcity-openstack-plugin/issues/2

### CODE REVIEW

File: build/pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <artifactId>cloud-openstack</artifactId>
         <groupId>jetbrains.buildServer.clouds</groupId>
-        <version>1.3-SNAPSHOT</version>
+        <version>1.3</version>
```
Comment (axel3rd) [inline build/pom.xml:7]: Development version (-SNAPSHOT) should be kept, release will done on dedicated process.

```suggestion
        <version>1.3-SNAPSHOT</version>
```

File: cloud-openstack-agent/pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <artifactId>cloud-openstack</artifactId>
         <groupId>jetbrains.buildServer.clouds</groupId>
-        <version>1.3-SNAPSHOT</version>
+        <version>1.3</version>
```
Comment (axel3rd) [inline cloud-openstack-agent/pom.xml:7]: Development version (-SNAPSHOT) should be kept, release will done on dedicated process.

```suggestion
        <version>1.3-SNAPSHOT</version>
```

File: cloud-openstack-common/pom.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <artifactId>cloud-openstack</artifactId>
         <groupId>jetbrains.buildServer.clouds</groupId>
-        <version>1.3-SNAPSHOT</version>
+        <version>1.3</version>
```
Comment (axel3rd) [inline cloud-openstack-common/pom.xml:7]: Development version (-SNAPSHOT) should be kept, release will done on dedicated process.

```suggestion
        <version>1.3-SNAPSHOT</version>
```

File: cloud-openstack-server/pom.xml [hunk 4]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <artifactId>cloud-openstack</artifactId>
         <groupId>jetbrains.buildServer.clouds</groupId>
-        <version>1.3-SNAPSHOT</version>
+        <version>1.3</version>
```
Comment (axel3rd) [inline cloud-openstack-server/pom.xml:8]: Development version (-SNAPSHOT) should be kept, release will done on dedicated process.

```suggestion
        <version>1.3-SNAPSHOT</version>
```

File: pom.xml [hunk 5]
```diff
@@ -3,7 +3,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>jetbrains.buildServer.clouds</groupId>
     <artifactId>cloud-openstack</artifactId>
-    <version>1.3-SNAPSHOT</version>
+    <version>1.3</version>
```
Comment (axel3rd) [inline pom.xml:6]: Development version (-SNAPSHOT) should be kept, release will done on dedicated process.

```suggestion
        <version>1.3-SNAPSHOT</version>
```

**Comment by axel3rd** [comment]
@promiteyka : Many thanks for this pull request. Could the [SonarQube quality gate](https://sonarcloud.io/dashboard?id=jetbrains.buildServer.clouds%3Acloud-openstack&pullRequest=48) taken into account ?

About unit tests, are they successful locally or not ?

**Comment by promiteyka** [comment]
> @promiteyka : Many thanks for this pull request. Could the [SonarQube quality gate](https://sonarcloud.io/dashboard?id=jetbrains.buildServer.clouds%3Acloud-openstack&pullRequest=48) taken into account ?
> 
> About unit tests, are they successful locally or not ?

We did not change the tests, but we checked the modified plugin on our server and it works as expected. I propose to finalize the code as part of this pull request.

**Comment by promiteyka** [comment]
@axel3rd 
The first build - https://teamcity.jetbrains.com/viewLog.html?buildId=2975379&buildTypeId=TeamCityThirdPartyPlugins_OpenStackCloudSupport_PullRequestAnalysis the tests failed because of 500 errors on devstack
the second build finished successfully https://teamcity.jetbrains.com/viewLog.html?buildId=2975435&buildTypeId=TeamCityThirdPartyPlugins_OpenStackCloudSupport_PullRequestAnalysis without any errors.

Now we are working with sonarqube errors.

File: build/pom.xml [hunk 6]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <artifactId>cloud-openstack</artifactId>
         <groupId>jetbrains.buildServer.clouds</groupId>
-        <version>1.3-SNAPSHOT</version>
+       <version>1.3-SNAPSHOT</version>
```
Comment (axel3rd) [inline build/pom.xml:7]: If some code change (SonarQube quality flaw, ...), a space ` <version>1.3-SNAPSHOT</version>` could perhaps be added (x5 files), to not introduce some changes if there are not, better for commit visibility :wink:.

**Comment by promiteyka** [comment]
We tested the updated plugin on versions 2019.2.4 and 2020.1. We installed it on production servers.

**Comment by axel3rd** [comment]
> We tested the updated plugin on versions 2019.2.4 and 2020.1. We installed it on production servers.

@promiteyka: Thanks for work & tests

Some questions:

1. Can you *squash* all PR commits in one (Comment sample: `Fix #2 instance removing while profile changes`) ? if I do it with GitHub option during merge/rebase, I don't know what it append with multiple-committers.
1. The [SonarQube quality gate](https://sonarcloud.io/dashboard?id=jetbrains.buildServer.clouds%3Acloud-openstack&pullRequest=48) is currently **red** due **no (new) unit-tests** (and not introducing new *Code Smells* would be great). To have sustainable plugin, is it relevant to achieve that ? I can rebase/merge without it, but I will not do the 1.3 release without take time to fix it.

Let me know for this 2 points.

----

For informations/check, executing unit test [OpenstackCloudClientTest.testV3](https://github.com/yandex-qatools/teamcity-openstack-plugin/blob/master/cloud-openstack-server/src/test/java/jetbrains/buildServer/clouds/openstack/OpenstackCloudClientTest.java#L65) on some Openstack platform where some existing VM in tenant and where the stop process is a little long, I can see:

```
Retrieve floating ip for future instance association
Floating ip: 10.0.0.1
Creating openstack instance with [removed]
Associating floating ip to serverId 360a0989-77f1-4237-b619-d08680249953
(Waiting fixed ip before floating ip association on serverId: 360a0989-77f1-4237-b619-d08680249953)
Restore instances of openstack image id=377f47ef-f656-4349-a135-0e72fce24c1e ...
Cloud instance restored: id=360a0989-77f1-4237-b619-d08680249953, name=openstack-test-teamcity-plugin-1592390169863, status=RUNNING
Pinging openstack-test-teamcity-plugin-1592390178908 for status
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390178908', result is 'ACTIVE' (previous internal status setted was 'Running')
Pinging openstack-test-teamcity-plugin-1592390169863 for status
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390169863', result is 'ACTIVE' (previous internal status setted was 'Starting')
2020-06-17 12:36:20 INFO  clouds.:91 - Stopping cloud openstack instance openstack-test-teamcity-plugin-1592390169863
Pinging openstack-test-teamcity-plugin-1592390178908 for status
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390178908', result is 'ACTIVE' (previous internal status setted was 'Running')
Pinging openstack-test-teamcity-plugin-1592390169863 for status
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390169863', result is 'ACTIVE' (previous internal status setted was 'Scheduled to stop')
Pinging openstack-test-teamcity-plugin-1592390178908 for status
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390178908', result is 'ACTIVE' (previous internal status setted was 'Running')
Pinging openstack-test-teamcity-plugin-1592390169863 for status
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390169863', result is 'ACTIVE' (previous internal status setted was 'Stopping')
Pinging openstack-test-teamcity-plugin-1592390178908 for status
# Multiple Running/Stopping phase
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390178908', result is 'ACTIVE' (previous internal status setted was 'Running')
Pinging openstack-test-teamcity-plugin-1592390169863 for status
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390169863', result is 'ACTIVE' (previous internal status setted was 'Stopping')
Pinging openstack-test-teamcity-plugin-1592390178908 for status
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390178908', result is 'SHUTOFF' (previous internal status setted was 'Running')
2020-06-17 12:37:23 INFO  clouds.:91 - Terminating cloud openstack instance openstack-test-teamcity-plugin-1592390178908
Pinging openstack-test-teamcity-plugin-1592390169863 for status
Getting instance status from openstack for 'openstack-test-teamcity-plugin-1592390169863', result is 'SHUTOFF' (previous internal status setted was 'Stopping')
2020-06-17 12:37:24 INFO  clouds.:91 - Terminating cloud openstack instance openstack-test-teamcity-plugin-1592390169863
2020-06-17 12:37:25 INFO  clouds.:91 - OpenstackCloudImage.dispose ...
```

The process **pings 2 different instances**, but only one is existing in Openstack platform. Don't know if the problem comes from Id usage or restore from existing VMs.

**Comment by axel3rd** [comment]
@promiteyka : Please see PR [#1](https://github.com/promiteyka/teamcity-openstack-plugin/pull/1) on your repo about my previous comments.

**Comment by promiteyka** [comment]
About a squash commit, should I reopen a new pull request with all the changes in one commit?

**Comment by axel3rd** [comment]
> About a squash commit, should I reopen a new pull request with all the changes in one commit?

No, you can do it directly on branch linked to pull request, by using a *force push* (after the squash).

If master is used (ie like the current: `promiteyka:master`), perhaps allowing *force push* on `master` branch could be required.

**Comment by axel3rd** [comment]
I have reintroduced [a potential bug](https://sonarcloud.io/project/issues?id=jetbrains.buildServer.clouds%3Acloud-openstack&issues=AXLbWG34wbLAILE6r7c5&open=AXLbWG34wbLAILE6r7c5&pullRequest=48) in my commit [fefe6109d9f4c012bab10bcb3590f87d76cdf5f2](https://github.com/yandex-qatools/teamcity-openstack-plugin/pull/48/commits/fefe6109d9f4c012bab10bcb3590f87d76cdf5f2)  :confounded: :thumbsdown: :facepunch:

**Comment by promiteyka** [comment]
Squash commit is done.

**Comment by promiteyka** [comment]
> I have reintroduced [a potential bug](https://sonarcloud.io/project/issues?id=jetbrains.buildServer.clouds%3Acloud-openstack&issues=AXLbWG34wbLAILE6r7c5&open=AXLbWG34wbLAILE6r7c5&pullRequest=48) in my commit [fefe6109d9f4c012bab10bcb3590f87d76cdf5f2](https://github.com/yandex-qatools/teamcity-openstack-plugin/pull/48/commits/fefe6109d9f4c012bab10bcb3590f87d76cdf5f2) 😖 👎 👊

What do we do next?

**Comment by axel3rd** [comment]
> Squash commit is done.

@promiteyka: Many thanks. A remark about [this commit](https://github.com/promiteyka/teamcity-openstack-plugin/commit/6803996a345811a3c28f37274faa68301c07918c), the author is me due to my last PR (`axel3rd authored and Dmitry.Vasilyev committed on 18 Feb`). If we want you become contributor of this plugin with this commit, I think you must amend it (`git commit --amend --author="John Doe <john@doe.org>"`, with correct name & email using for for GitHub pseudo).

----

> What do we do next?

If you want take in attention my previous comment (the amend) ; you can quickly fix it (fix, squash 2 last commit ; it will fix the amend) and I would thank you ; otherwise I will fix it after the rebase/merge of this PR.

----

Let me know your preferences

**Comment by promiteyka** [comment]
@axel3rd 
I apologize for the delay, was on a short vacation. 
It is not important for me to be a contributor; for me, just a working plugin is enough.
Thank you.

**Pull request merged**